packages/vue/src/pages/planning-scheduler.vue

```diff
@@ -0,0 +1,378 @@
+<template>
+  <div class="planning-scheduler">
+    <div class="view-wrapper">
+      <div class="panels">
+        <left-side-panel class="left-side-panel">
+          <div class="left-side-panel-content">
+            <div class="buttons">
+              <dx-button
+                text="Today"
+                @click="onSelectedDateChange"
+              />
+              <dx-button
+                text="Create event"
+                type="default"
+                @click="showAppointmentCreationForm"
+              />
+            </div>
+            <div class="calendar">
+              <dx-calendar
+                :value="currentDate"
+                @value-changed="onCalendarDateChange"
+              />
+            </div>
+            <calendar-list
+              :data-source="defaultCalendarListItems"
+              @list-selection-changed="calendarListChanged"
+            />
+          </div>
+        </left-side-panel>
+
+        <div class="scheduler-container">
+          <dx-scheduler
+            ref="schedulerRef"
+            height="inherit"
+            :adaptivity-enabled="!!screenInfo.isXSmall"
+            :all-day-panel-mode="'hidden'"
+            :data-source="tasks"
+            :current-view="currentView"
+            :current-date="schedulerCurrentDate"
+            :start-day-hour="4"
+            :show-current-time-indicator="false"
+            @option-changed="onSchedulerOptionChange"
+            @appointment-click="onAppointmentClick"
+            @appointment-tooltip-showing="onAppointmentTooltipShowing"
+            @cell-click="onCellClick"
+          >
+            <dx-resource
+              field-expr="calendarId"
+              label="Calendar"
+              :data-source="resourcesList"
+            />
+            <dx-view type="day" />
+            <dx-view type="workWeek" />
+            <dx-view type="month" />
+            <dx-view type="agenda" />
+          </dx-scheduler>
+          <dx-tooltip
+            ref="tooltipRef"
+            show-event="click"
+            :target="selectedAppointment?.target"
+            :position="tooltipPosition"
+          >
+            <scheduler-tooltip
+              :selected-appointment-data="selectedAppointment?.data"
+              @click-edit-appointment="editSelectedAppointment"
+              @click-delete-appointment="deleteSelectedAppointment"
+            />
+          </dx-tooltip>
+          <dx-speed-dial-action
+            icon="add"
+            :visible="screenInfo.isXSmall"
+            @click="showAppointmentCreationForm"
+          />
+        </div>
+
+        <right-side-panel
+          class="right-side-panel"
+          :is-opened="isRightPanelOpen"
```

Comment:
If the right panel is opened when the Month view is selected, should it be closed automatically if the view is changed back?

Reply:
it's approved by designer